### ISSUE TITLE
Wide Neural Networks of Any Depth Evolve as Linear Models Under Gradient Descent

### ISSUE THREAD

**Issue by icoxfog417** [arXivTimes/arXivTimes#1108]
## 一言でいうと

幅が無限とした場合のDNNを最急降下法で学習させる処理は、線形変換と同等であるとした論文(=学習結果は一次のテイラー展開で置き換えられる)。損失関数が2次の場合は出力がGaussianで維持されるため学習の過程はGaussian Processと見做すことができ、この点がBayesianNetと異なるとしている。

### 論文リンク

https://arxiv.org/abs/1902.06720

### 著者/所属機関

Jaehoon Lee, Lechao Xiao, Samuel S. Schoenholz, Yasaman Bahri, Jascha Sohl-Dickstein, Jeffrey Pennington

* Google Brain

### 投稿日付(yyyy/MM/dd)

2019/2/18

**Comment by icoxfog417** [comment]
実装が公開

https://github.com/google/neural-tangents

**Comment by icoxfog417** [comment]
Colabが公開。

https://colab.research.google.com/github/google/neural-tangents/blob/master/notebooks/neural_tangents_cookbook.ipynb#scrollTo=LKnYe66NVzFF

**Comment by icoxfog417** [comment]
解説記事
https://rajatvd.github.io/NTK/

**Comment by Hiroki11x** [comment]
- パラメータ空間のダイナミクス 

我々は、パラメータ空間における広いネットワークの学習ダイナミクスが、すべてのネットワークパラメータ、重み、バイアスの集合においてアフィン系であるモデルの学習ダイナミクスと等価であることを示す。

この結果は、損失関数の選択に関係なく保持されます。

二乗損失については、時間の関数として閉形解が得られます。


- 線形化のための十分な条件。
隠れ層の幅であるnが十分に大きい場合、⌘criticalよりも小さい学習率を持つ勾配降下訓練軌道が線形化されたネットワークの軌道のO n1/2 -近傍に留まるような閾値学習率⌘criticalが存在することを正式に証明します（定理2.1参照）。


- 出力分布のダイナミクス。ニューラルネットワークの予測が 勾配降下訓練全体を通して, 幅が無限大になるにつれてGPによって記述されます.
定理2.2)は，Jacotら[13]の結果を拡張したものである．

さらに、学習中のこのGPの進化の時間依存性の表現を明示的に導出する。最後に、この結果の新しい解釈を提供する。

特に、この結果は、勾配降下がパラメータのベイズ的事後サンプリングと異なるメカニズムを定量的に理解することを提供する。
どちらの手法もGPからドローを生成しますが、勾配降下法は確率モデルの事後サンプリングを生成しません。


- 大規模な実験的支持。
我々は、有限幅の設定における理論の適用可能性を経験的に調査している。

我々は、広残差ネットワーク[14]のような実用的なネットワークアーキテクチャを含む様々な条件で、学習ダイナミクスと事後関数分布の両方を正確に特徴付けることができることを発見した。



- パラメータ化の独立性。
以前の研究では後者を仮定していたが、線形化の結果は標準パラメタリゼーションとNTKパラメタリゼーションの両方で維持され、その効果は特定のパラメタリゼーションではなく幅の増加によるものであることが強調されていることに注意する。

- 解析的ReLUとerfニューラルタンジェントカーネル. ReLUまたはerf非線形性を持つ完全連結ネットワークに対応する解析的ニューラルタンジェントカーネルを計算する。

www.DeepL.com/Translator（無料版）で翻訳しました。

**Comment by Hiroki11x** [comment]
本研究では、勾配降下下での幅の広いニューラルネットワークの学習ダイナミクスを調べ、そのダイナミクスの重み空間記述が驚くほど単純になることを発見した。

幅が大きくなると、ニューラルネットワークは、初期化時のパラメータに対して1次のテイラー展開で効果的に置き換えることができる。


この線形モデルでは、勾配降下のダイナミクスが解析的に扱いやすくなります。

線形化は無限の幅の極限でのみ正確であるが、それにもかかわらず、有限の幅の構成においても、元のネットワークの予測値と線形化されたバージョンの予測値の間に優れた一致が見られる。

この一致は、異なる アーキテクチャ、最適化手法、および損失関数についての情報を提供します。


二乗損失については、厳密な学習ダイナミクスは閉形式の解を認めており、GPの観点から予測分布の進化を特徴付けることができます。


この結果は 本研究では、"sample-then-optimize "事後サンプリング[12]をディープニューラルネットワークの訓練に拡張しました。


我々の経験的なシミュレーションでは，この結果が 異なるランダムな初期化を持つ有限幅モデルのアンサンブル。

**Comment by Hiroki11x** [comment]
ディープラーニング研究の長年の目標は、訓練と一般化を正確に特徴づけることでした。しかし、ニューラルネットワークの損失ランドスケープは複雑であることが多く、学習ダイナミクスの理論がつかみにくいものとなっていました。それにもかかわらず、大きな幅の限界では、ニューラルネットワークは解答可能なダイナミクスを持つ線形モデルとして進化します。

a. パラメータ空間のダイナミクス。
パラメータ空間のダイナミクス：パラメータ空間における幅の広いネットワークの学習ダイナミクスは、すべてのネットワークパラメータの集合がアフィンであるモデルの学習ダイナミクスと等価である。

b. 線形化のための十分な条件。
b. 線形化のための十分な条件：閾値学習率η(臨界)が存在し、その閾値よりも小さい学習率を持つニューラルネットワークの勾配降下訓練は、大きな幅の線形化によって十分に近似される。

c. 出力分布のダイナミクス。
勾配降下訓練中のニューラルネットワークの予測は、幅が無限大になるにつれて弱くGPに収束していく。
このGPの時間依存性の表現を導出し、ベイズの事後GPとの違いに注目する。

d. 論文の続きを読む
線形化のパラメータ化独立性
モメンタム、非二乗損失（クロスエントロピーなど 
ErfとReluのNTKの解析式
コラボチュートリアル

**Comment by icoxfog417** [comment]
解説記事
https://www.inference.vc/neural-tangent-kernels-some-intuition-for-kernel-gradient-descent/
